daisy/resource_registry.go

```diff
@@ -197,7 +197,11 @@ func (r *baseResourceRegistry) regURL(url string) (*Resource, DError) {
 		if err != nil {
 			return nil, err
 		}
-		return nil, typedErrf(resourceDNEError, "%s does not exist", url)
+		errStr := "%s does not exist"
+		if strings.HasSuffix(url, "default") {
+			errStr += ". please provide a specific resource name."
```

Comment:
While an improvement, this still doesn't tell customer what to do (provide --network flag in this case).

Can we catch this exception in wrappers, detect it's network related and provide a more specific error message?

Reply:
We can't, unless with a small refactoring. Let me try and propose in the next commit.